src/app/components/normal-component/lobby/lobby.component.spec.ts

```diff
@@ -25,39 +28,426 @@ describe('LobbyComponent', () => {
+                expectValidRouting(router, ['/play', 'Quarto', startedPartUserDoNotPlay.id], OnlineGameWrapperComponent);
+            }));
+        });
+        describe('as a player', () => {
+            beforeEach(() => {
+                // Given an user observing a part as a Player
+                ConnectedUserServiceMock.setObservedPart(MGPOptional.of({
+                    id: startedPartUserPlay.id,
+                    role: 'Player',
+                    typeGame: 'P4',
+                }));
+            });
+            it('should forbid user to join another game', fakeAsync(async() => {
+                // And a lobby where the a part user do not play is present
+                setLobbyPartList([startedPartUserDoNotPlay]);
+                testUtils.detectChanges();
+
+                // When clicking on the part
+                spyOn(component.messageDisplayer, 'criticalMessage').and.callThrough();
+                await testUtils.clickElement('#part_0');
+                tick(3000); // 3 sec of toast display
+
+                // Then the refusal reason should be given
+                const reason: string = GameActionFailure.YOU_ARE_ALREADY_PLAYING();
+                expect(component.messageDisplayer.criticalMessage).toHaveBeenCalledOnceWith(reason);
+            }));
+            it('should allow users to play their games from several tabs', fakeAsync(async() => {
+                // And a lobby where the part player plays is present
+                setLobbyPartList([startedPartUserPlay]);
+                const router: Router = TestBed.inject(Router);
+                spyOn(router, 'navigate').and.resolveTo();
+                testUtils.detectChanges();
+
+                // When clicking on the part
+                await testUtils.clickElement('#part_0');
+
+                // Then we should have went to see the game
+                expectValidRouting(router, ['/play', 'Quarto', startedPartUserPlay.id], OnlineGameWrapperComponent);
+            }));
+        });
+        describe('as an observer', () => {
+            beforeEach(() => {
+                // Given an user observing a part as an Observer
+                ConnectedUserServiceMock.setObservedPart(MGPOptional.of({
+                    id: startedPartUserDoNotPlay.id,
+                    role: 'Observer',
+                    typeGame: 'P4',
+                }));
+            });
+            it('should allow to observe a second part', fakeAsync(async() => {
+                // And a lobby where another started part that user do not play is present
+                setLobbyPartList([anotherStartedPartUserDoNotPlay]);
+                const router: Router = TestBed.inject(Router);
+                spyOn(router, 'navigate').and.resolveTo();
+                testUtils.detectChanges();
+
+                // When clicking on the part
+                await testUtils.clickElement('#part_0');
+
+                // Then we should have went to see the game
```

Comment:
à voir comment mieux formuler les choses, mais le mélange « given a user » et « we ...» ça correspond pas trop